policies/testing.md

```diff
@@ -0,0 +1,42 @@
+Testing Policy
+==============
+
+This applies to all stable and development branches of the main code repository.
+Except where noted below:
+
+- All Pull Requests adding new functionality to the applications, libraries,
+providers or engines must include suitable tests.
+- All Pull Requests fixing a functional behavior defect in the applications,
+libraries, providers or engines must include a test for that defect.
+
+Pull Requests that do not change the functional behaviour of the applications,
+libraries, providers or engines do not require tests to be added. For example
+the following types of changes do not require tests:
+ - Changes to comments, formatting, internal naming or similar
+ - Fixes for performance
+ - Refactoring
```

Comment:
Refactoring seems to be a good time to check that corner cases are tested. It might not need new tests, but it seems to be a good time to check that there are enough tests.

Reply:
I've added a recommendation to this effect.